### ISSUE TITLE
Migration from older (pre 2.0) blog failed

### ISSUE THREAD

**Issue by prembo** [silverstripe/silverstripe-blog#270]
I did a composer update yesterday to dev-master (070978e).  An existing blog that ran nicely on a pre 2.0 blog (8b28a1d) didn't migrate properly.  Please note that on the first `dev/build` the new `Blog` & `BlogPost` types were created, but only on the second `dev/build` did the output say:

```
Migrating legacy blog records
Migrating 38 legacy BlogEntry records in stage Stage.
```

Having said that though, the blog posts appear under Blog in the SiteTree. What previously was the `BlogHolder` page was converted to a `Blog` page.  

However all the posts, previously `BlogEntry`, have not been successfully converted to `BlogPost`.  Also none of the fields such as Tags, Author or Date Published have migrated across.  :-(

Also, for some reason, when I manually paste a comma separated line of tags into the tag field in the CMS admin, the field doesn't accept this and seems to display multiple entries in the list of the same tag.
![screen shot 2015-06-08 at 11 16 31 am](https://cloud.githubusercontent.com/assets/1043612/8026945/e24ffe9c-0dcf-11e5-8868-1c28b023dfc9.png)

**Comment by prembo** [comment]
The multiple tags problem has its own issue: #271

**Comment by prembo** [comment]
I'm not sure why, but as I've continued on working after a few `dev/build?flush=all`, all the blog posts now have been converted to `BlogPost`.  

Strange, but on the first `dev/build` it said:
```
Migrating legacy blog records
Migrating 38 legacy BlogEntry records in stage Stage.
```

On the second `dev/build` it said that 14 BlogEntry items were migrated:
```
Migrating legacy blog records
Migrating 14 legacy BlogEntry records in stage Stage.
```

And on subsequent `dev/build`'s there was no mention of any records needing migration.  But it has only been after a few more `dev/build`'s that I've finally noticed that the former `BlogEntry` posts which I had been manually updating / fixing, had the proper `BlogPost` page type, and also that the tags and author field was appropriately populated.  Earlier they were empty and there wasn't a Page Type assigned.

So for some reason, it is all working again.  Maybe the secret is to do many `dev/build?flush=all`'s?
